src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerContentFacetImpl.java

```diff
@@ -92,18 +93,18 @@ public Content get(final String path) throws IOException {
   }
 
   @Override
-  public Content put(final String path, final Content content, final AssetKind assetKind) throws IOException {
+  public Content put(final String path, final Payload payload, final AssetKind assetKind) throws IOException {
     StorageFacet storageFacet = facet(StorageFacet.class);
-    try (TempBlob tempBlob = storageFacet.createTempBlob(content, hashAlgorithms)) {
+    try (TempBlob tempBlob = storageFacet.createTempBlob(payload, hashAlgorithms)) {
       switch (assetKind) {
         case ZIPBALL:
-          return doPutContent(path, tempBlob, content, assetKind);
+          return doPutContent(path, tempBlob, payload, assetKind);
         case PACKAGES:
```

Comment:
> Trivial: PACKAGES, LIST, PROVIDER could be under the same case statement.

Depends on how you feel about drop-through case statements. In my opinion they're a bad idea, so I generally tend to avoid them, and I'm not sure our internal coding standards have a preference or guideline one way or the other.